packages/mgt-chat/src/statefulClient/GraphNotificationUserClient.ts

```diff
@@ -199,93 +197,117 @@ export class GraphNotificationUserClient {
     if (!subscription?.notificationUrl) throw new Error('Subscription not created');
     log(subscription);
 
-    const awaits: Promise<void>[] = [];
-    // Cache the subscription in storage for re-hydration on page refreshes
-    awaits.push(this.cacheSubscription(subscription));
+    await this.cacheSubscription(subscription);
 
     // create a connection to the web socket if one does not exist
-    if (!this.connection) awaits.push(this.createSignalRConnection(subscription.notificationUrl));
+    if (!this.connection) await this.createSignalRConnection(subscription.notificationUrl);
 
     log('Invoked CreateSubscription');
-    return Promise.all(awaits);
   }
 
-  private readonly startRenewalTimer = () => {
-    if (this.renewalInterval !== undefined) this.timer.clearInterval(this.renewalInterval);
-    this.renewalInterval = this.timer.setInterval(this.syncTimerWrapper, appSettings.renewalTimerInterval * 1000);
-    log(`Start renewal timer . Id: ${this.renewalInterval}`);
+  private readonly renewalSync = () => {
+    void this.renewal();
   };
 
-  private readonly syncTimerWrapper = () => void this.renewalTimer();
-
-  private readonly renewalTimer = async () => {
-    log(`running subscription renewal timer for userId: ${this.userId} sessionId: ${this.sessionId}`);
+  private readonly renewal = async () => {
```

Comment:
This is so much better, thank you.

As discussed, generally I would wrap the whole renewal code in a try...catch except the setTimeout so that the interval continued processing instead of throwing unhandled exceptions, but I will accept this as the PG code doesn't do a try...catch either.